### PR TITLE
Modernize default upgrade policy patch

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
The upgrade policy change needs to reboot the machine, which messes with systemd’s first boot.

Back when this was implemented, we wanted this patch to afect pre-existing distros. Any pre-existing distro will have received the patch by now, so we no longer need this requirement.

This’ll also speed up install time.

UDENG-268

### CODE REVIEW

File: DistroLauncher/Patch.cpp [hunk 1]
```diff
@@ -44,6 +44,49 @@ namespace Ubuntu::PatchingFunctions
...
+            if (starts_with(distro, L"Ubuntu") && ends_with(distro, L"LTS")) {
+                return "never";
+            }
+            return "normal";
+        }();
+
+        // Copy until "Prompt=" is found
+        std::string line;
+        std::string_view prefix;
+        auto it = getline(unused, line);
```
[the remark this answers is not in the document]
Comment (EduardGomezEscandell) [reply DistroLauncher/Patch.cpp:65]: `it` seems like a perfectly fine name for an iterator.

[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply DistroLauncher/Patch.cpp:65]: I agree. Let's add it to the `.clang-tidy`'s allow list for variable and parameter names:

```
CheckOptions:
  - key: readability-identifier-length.IgnoredParameterNames
    value: "hr|ok|st"
  - key: readability-identifier-length.IgnoredVariableNames
    value: "hr|ok|st"
```

File: DistroLauncher/Patch.cpp [hunk 2]
```diff
@@ -44,6 +44,49 @@ namespace Ubuntu::PatchingFunctions
         conf << "[Unit]\nConditionVirtualization=!container\n";
         return !conf.fail();
     }
+
+    bool SetDefaultUpgradePolicy(std::istreambuf_iterator<char> unused, std::ostream& output)
```
Comment (CarlosNihelton) [inline DistroLauncher/Patch.cpp:48]: A minor thing: `unused` is being used, so we should assign a more meaningful name.

**Review by CarlosNihelton — CHANGES_REQUESTED** [review]
Two observations only:

- the `it` naming 
- the `unused` parameter being used.

**Review by CarlosNihelton — APPROVED** [review]
LGTM!

**Comment by EduardGomezEscandell** [comment]
Force-pushed to rebase main. Will merge if/when tests pass.

**Pull request merged**